src/server/service/kv.rs

```diff
@@ -1138,6 +767,996 @@ impl<T: RaftStoreRouter + 'static, E: Engine> tikvpb_grpc::Tikv for Service<T, E
+                );
+            } else {
+                notifier.notify();
+            }
+        }
+        Ok(())
+    });
+    executor.spawn(f);
+}
+
+fn handle_batch_commands_request<E: Engine>(
+    storage: &Storage<E>,
+    cop: &Endpoint<E>,
+    peer: String,
+    executor: &TaskExecutor,
+    id: u64,
+    req: BatchCommandsRequest_Request_oneof_cmd,
+    tx: Sender<(u64, BatchCommandsResponse_Response)>,
+    thread_load: Arc<ThreadLoad>,
+    heavy_load_threshold: usize,
+) {
+    match req {
+        BatchCommandsRequest_Request_oneof_cmd::Get(req) => {
+            let timer = GRPC_MSG_HISTOGRAM_VEC.kv_get.start_coarse_timer();
+            let resp = future_get(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::Get)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.kv_get.inc());
+            response_batch_commands_request(
+                executor,
+                id,
+                resp,
+                tx,
+                timer,
+                thread_load,
+                heavy_load_threshold,
+            );
+        }
+        BatchCommandsRequest_Request_oneof_cmd::Scan(req) => {
+            let timer = GRPC_MSG_HISTOGRAM_VEC.kv_scan.start_coarse_timer();
+            let resp = future_scan(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::Scan)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.kv_scan.inc());
+            response_batch_commands_request(
+                executor,
+                id,
+                resp,
+                tx,
+                timer,
+                thread_load,
+                heavy_load_threshold,
+            );
+        }
+        BatchCommandsRequest_Request_oneof_cmd::Prewrite(req) => {
+            let timer = GRPC_MSG_HISTOGRAM_VEC.kv_prewrite.start_coarse_timer();
+            let resp = future_prewrite(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::Prewrite)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.kv_prewrite.inc());
+            response_batch_commands_request(
+                executor,
+                id,
+                resp,
+                tx,
+                timer,
+                thread_load,
+                heavy_load_threshold,
+            );
+        }
+        BatchCommandsRequest_Request_oneof_cmd::Commit(req) => {
+            let timer = GRPC_MSG_HISTOGRAM_VEC.kv_commit.start_coarse_timer();
+            let resp = future_commit(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::Commit)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.kv_commit.inc());
+            response_batch_commands_request(
+                executor,
+                id,
+                resp,
+                tx,
+                timer,
+                thread_load,
+                heavy_load_threshold,
+            );
+        }
+        BatchCommandsRequest_Request_oneof_cmd::Import(_) => {
+            panic!("unimplemented");
```

Comment:
should it be implemented now?

Reply:
I think we don't need to support it now. 
Btw, use `unimplemented!()` instead